compiling fails to ATParser.cpp:328

Got this:

```
mbed compile --build .build/K64F_iar_arm_default/ -m K64F -t IAR -c 
Building project mbed-os-cliapp (K64F, IAR)
Compile: at24mac.cpp
Compile: driverAtmelRFInterface.cpp
Compile: driverRFPhy.c
Compile: ESP8266Interface.cpp
Compile: ATParser.cpp
[Error] ATParser.cpp@328: [Pe029]: expected an expression
      _oobs.push_back((struct oob){strlen(prefix), prefix, cb});
                                  ^
"Z:\builds\ws\ARMmbed\mbed-os-cliapp\nsapi_changes\mbed-os-cliapp\esp8266-driver\ESP8266\ATParser\ATParser.cpp",328  Error[Pe029]: expected an expression
[ERROR] 
      _oobs.push_back((struct oob){strlen(prefix), prefix, cb});
                                  ^
"Z:\builds\ws\ARMmbed\mbed-os-cliapp\nsapi_changes\mbed-os-cliapp\esp8266-driver\ESP8266\ATParser\ATParser.cpp",328  Error[Pe029]: expected an expression

[mbed ERROR] "python" returned error code 1.
```

--- a/ESP8266/ATParser/ATParser.cpp
+++ b/ESP8266/ATParser/ATParser.cpp
@@ -325,5 +325,9 @@ bool ATParser::recv(const char *response, ...)
 // oob registration
 void ATParser::oob(const char *prefix, Callback<void()> cb)
 {
-    _oobs.push_back((struct oob){strlen(prefix), prefix, cb});
+    struct oob oob;
+    oob.len = strlen(prefix);
+    oob.prefix = prefix;
+    oob.cb = cb;
+    _oobs.push_back(oob);
 }

